### PR TITLE
adding new verification test

### PR DESCRIPTION
**Pull request by ekiernan**
This PR adds a new verification task to compare Optimus library metric CSV files. If the md5sums fail, it performs additional checks on the lines that are not related to doublet score.

### CODE REVIEW

**Comment by ekiernan** [comment]
Testing Optimus scientific here: https://gotc-jenkins.dsp-techops.broadinstitute.org/job/warp-workflow-tests/85996/

**Comment by ekiernan** [comment]
retest this please

**Review by nikellepetrillo — APPROVED** [review]
looks good, thank you!

File: verification/VerifyTasks.wdl [hunk 1]
```diff
@@ -613,3 +613,61 @@ task CompareSnapTextFiles {
...
+  }
+
+  command {
+    exit_code=0
+
+    test_files_length=~{length(test_text_files)}
+    truth_files_length=~{length(truth_text_files)}
+    if [ $test_files_length -ne $truth_files_length ]; then
+      exit_code=1
+      echo "Error: Different number of input files ($test_files_length vs. $truth_files_length).  This is really not OK"
```
Comment (jessicaway) [inline verification/VerifyTasks.wdl:629]: Should we exit? No need to look further, right?

**Review by jessicaway — APPROVED** [review]
Looks good, thanks @ekiernan!

File: verification/VerifyTasks.wdl [hunk 2]
```diff
@@ -613,3 +613,53 @@ task CompareSnapTextFiles {
...
+        diff_output=$(diff <(grep -v -E $excluded_lines a.sorted) <(grep -v -E $excluded_lines b.sorted))
+
+        if [ -z "$diff_output" ]; then
+          echo "Files a.sorted and $b.sorted are the same when excluding specified lines."
+        else
+          echo "Files a.sorted and b.sorted have differences in non-excluded lines."
+          echo "$diff_output"
+          exit_code=2
+        fi
+      fi
```
Comment (jessicaway) [inline verification/VerifyTasks.wdl:654]:
```suggestion
    fi
```

File: verification/VerifyTasks.wdl [hunk 3]
```diff
@@ -613,3 +613,53 @@ task CompareSnapTextFiles {
...
+    sort "$a" > "a.sorted"
+    sort "$b" > "b.sorted"
+
+    echo "Calculating md5sums for $a and $b"
+    md5_a=$(md5sum "a.sorted" | cut -d ' ' -f1)
+    md5_b=$(md5sum "b.sorted" | cut -d ' ' -f1)
+
+    if [ $md5_a = $md5_b ]; then
+        echo "Files $a.sorted and $b.sorted have matching md5sums and are the same."
+      else
```
Comment (jessicaway) [inline verification/VerifyTasks.wdl:638]:
```suggestion
    else
```

Comment (jessicaway) [reply verification/VerifyTasks.wdl:638]: Being super nit picky, but I think the indentation is a bit off (sometimes using 4 spaces and sometimes 2 spaces)

**Review by jessicaway — APPROVED** [review]
Looks good @ekiernan, thank you!

**Review by nikellepetrillo — APPROVED** [review]
looks good, ty liz!

File: verification/VerifyTasks.wdl (hunk 2 above)
Comment (ekiernan) [reply verification/VerifyTasks.wdl:653]: Thank you! I reformatted.

**Pull request merged**